[quant] making quant utilities inplace

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25054 [quant] making quant utilities inplace**

Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

Commit:
[quant] making quant utilities inplace

Commit:
Update on "[quant] making quant utilities inplace"


Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -240,15 +239,11 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING):
-    Return:
-        A quantized model
     """
     model.eval()
```
can we do a deepcopy of model before this line

zafartahirov replied:
Wait, didn't we decide not to deepcopy?

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -260,21 +255,18 @@ def quantize_dynamic(model, qconfig_dict=DEFAULT_QCONFIG_DICT, mapping=DEFAULT_D
 def quantize_qat(model, run_fn, run_args):
     r"""Do quantization aware training and output a quantized model
     """
     model.train()
```
same for this

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -283,26 +275,21 @@ def convert(module, mapping=DEFAULT_MODULE_MAPPING):
-            reassign[name] = new_mod
+        if type(mod) not in SWAPPABLE_MODULES:
+            convert(mod, mapping)
+        swapped_mod = swap_module(mod, mapping)
```
nit: merge these two lines into one

Commit:
Update on "[quant] making quant utilities inplace"


Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -240,12 +241,11 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING):
             function that simply runs the prepared model or a training loop
         run_args: positional arguments for `run_fn`
-
-    Return:
```
this still returns a quantized model

jerryzh168 reviewed (approved):
please fix comments

Commit:
Update on "[quant] making quant utilities inplace"


Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

Commit:
Update on "[quant] making quant utilities inplace"


Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

Commit:
Update on "[quant] making quant utilities inplace"


Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

Commit:
Update on "[quant] making quant utilities inplace"


Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)
